forui/lib/src/widgets/rating/rating.dart

```diff
@@ -0,0 +1,183 @@
+
+  /// {@macro forui.foundation.doc_templates.autofocus}
+  final bool autofocus;
+
+  /// {@macro forui.foundation.doc_templates.focusNode}
+  final FocusNode? focusNode;
+
+  /// {@macro forui.foundation.doc_templates.onFocusChange}
+  final ValueChanged<bool>? onFocusChange;
+
+  /// The spacing between rating icons.
+  final double spacing;
+
+  /// Creates a [FRating] widget.
+  const FRating({
+    super.key,
+    this.value = 0.0,
+    this.count = 5,
+    this.onStateChanged,
+    this.enabled = true,
+    this.style,
+    this.filledIcon = const Icon(FIcons.star, color: Color(0xFFFFD700)), // Gold
+    this.emptyIcon = const Icon(FIcons.starOff, color: Color(0xFFBDBDBD)), // Gray
+    this.halfFilledIcon,
+    this.semanticsLabel,
+    this.autofocus = false,
+    this.focusNode,
+    this.onFocusChange,
+    this.spacing = 4.0,
+  });
+
+  @override
+  State<FRating> createState() => _FRatingState();
+
+  @override
+  void debugFillProperties(DiagnosticPropertiesBuilder properties) {
+    super.debugFillProperties(properties);
+    properties
+      ..add(DoubleProperty('value', value))
+      ..add(IntProperty('count', count))
+      ..add(FlagProperty('enabled', value: enabled, ifTrue: 'enabled'))
+      ..add(DiagnosticsProperty('style', style))
+      ..add(ObjectFlagProperty<ValueChanged<double>?>.has('onStateChanged', onStateChanged))
+      ..add(StringProperty('semanticsLabel', semanticsLabel))
+      ..add(DiagnosticsProperty<bool>('autofocus', autofocus))
+      ..add(DiagnosticsProperty<FocusNode?>('focusNode', focusNode))
+      ..add(ObjectFlagProperty<ValueChanged<bool>?>.has('onFocusChange', onFocusChange))
+      ..add(DoubleProperty('spacing', spacing));
+  }
+}
+
+class _FRatingState extends State<FRating> {
+  double _hoverValue = 0.0;
+  bool _isHovering = false;
+
+  double _calculateRating(double dx, double totalWidth) {
+    if (totalWidth <= 0) {
+      return 0;
+    }
+
+    final itemWidth = totalWidth / widget.count;
+    final x = dx.clamp(0.0, totalWidth);
+    final index = x ~/ itemWidth;
+    final localPosition = x - (index * itemWidth);
+
+    double rating = index + 1.0;
+    if (widget.halfFilledIcon != null && localPosition < itemWidth / 2) {
+      rating -= 0.5;
+    }
+
+    return rating.clamp(0.0, widget.count.toDouble());
+  }
+
+  void _handleHover(PointerHoverEvent event) {
+    final box = context.findRenderObject() as RenderBox?;
+    if (box == null) {
+      return;
+    }
+
+    final localPosition = box.globalToLocal(event.position);
+    setState(() => _hoverValue = _calculateRating(localPosition.dx, box.size.width));
+  }
+
+  void _handleTap(TapUpDetails details) {
+    final box = context.findRenderObject()! as RenderBox;
+    final localPosition = box.globalToLocal(details.globalPosition);
+    final rating = _calculateRating(localPosition.dx, box.size.width);
+
+    widget.onStateChanged?.call(rating);
+    setState(() => _isHovering = false);
+  }
```

Comment:
Generally, I think we should try to avoid using retrieving the renderbox like this. Perhaps, you can use the  [`PointerEnterEvent`](https://api.flutter.dev/flutter/gestures/PointerEnterEvent-class.html) passed into [`MouseRegion.onEnter`](https://api.flutter.dev/flutter/widgets/MouseRegion/onEnter.html) to calculate the size?

I think there's something similar for tap gestures too.